receiver/dcgmreceiver/scraper.go

```diff
@@ -216,14 +216,19 @@ func (s *dcgmScraper) pollClient(ctx context.Context, client *dcgmClient, metric
 			)/2,
 		)
 		s.settings.Logger.Sugar().Debugf("Waiting %s for the next collection", waitTime)
-		deviceMetrics := client.getDeviceMetrics()
-		select {
-		case <-ctx.Done():
-			return
-		case <-collectTriggerCh:
-			// Loop and trigger a collect() again.
-		case metricsCh <- deviceMetrics:
-		case <-time.After(waitTime):
+		after := time.After(waitTime)
+		for after != nil {
+			deviceMetrics := client.getDeviceMetrics()
+			select {
+			case <-ctx.Done():
+				return
+			case <-collectTriggerCh:
+				// Loop and trigger a collect() again.
+				after = nil
```

Comment:
Just for my own understanding, this is equivalent to:
```go
L:
		for {
			deviceMetrics := client.getDeviceMetrics()
			select {
			case <-ctx.Done():
				return
			case <-collectTriggerCh:
				// Loop and trigger a collect() again.
				break L
			case metricsCh <- deviceMetrics:
			case <-time.After(waitTime):
				break L
			}
		}
```
right?

Reply:
Almost, but your code constructs a new `time.After` channel each time it loops (potentially sleeping for more than the desired `waitTime`), whereas my code reuses the same one and will thus wait for only one instance of `waitTime`. I used my version just because loop labels look too much like goto labels and I have an aversion to them :)